### PR TITLE
chore: Upgrade amazon-cognito-identity-json Webpack & other changes

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

This change addresses our dev dependency dependabot alert on `json5`, `serialize-javascript`, and `ssri` by:
- Upgrading `amazon-cognito-identity-js` Webpack version to v4 (from v3)
- Removed unused Webpack plugins & set Terser config
- Removing unused Webpack dependencies in `pushnotification` package

#### Description of how you validated changes
Local build & test. Smoke tested `amazon-cognito-identity-js` UMD artifacts by executing user pool signUp & signIn.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,7 +75,6 @@
 		"babel-jest": "^24.9.0",
 		"babel-loader": "^8.3.0",
 		"codecov": "^3.6.5",
-		"compression-webpack-plugin": "^1.1.3",
```
Comment (jimblanc) [inline package.json:78]: Moved this off of the umbrella package as only `amazon-cognito-identity-js` appears to use it.

File: packages/amazon-cognito-identity-js/webpack.config.js [hunk 2]
```diff
@@ -28,27 +28,28 @@ var config = {
 		crypto: 'crypto',
 	},
 	plugins: [
-		new webpack.optimize.OccurrenceOrderPlugin(),
```
Comment (jimblanc) [inline packages/amazon-cognito-identity-js/webpack.config.js:31]: Default behavior, no longer needed.

**Review by cshfang — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (cshfang) [reply package.json:78]: This also appears to be true for `uglifyjs-webpack-plugin` below

File: packages/amazon-cognito-identity-js/package.json [hunk 3]
```diff
@@ -87,7 +87,9 @@
     "genversion": "^2.2.0",
     "react": "^16.0.0",
     "rimraf": "^2.5.4",
-    "uglifyjs-webpack-plugin": "^1.0.0",
-    "webpack": "^3.5.5"
+    "compression-webpack-plugin": "^1.1.12",
```
Comment (cshfang) [inline packages/amazon-cognito-identity-js/package.json:90]: Updating this to `^4.0.1` would also resolve additional related issues

**Review by israx — APPROVED** [review]

File: packages/amazon-cognito-identity-js/webpack.config.js [hunk 4]
```diff
@@ -1,6 +1,5 @@
-// version 3.11.0
-const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
-const CompressionPlugin = require('compression-webpack-plugin');
+// Webpack version: ^4.46.0
```
Comment (cshfang) [inline packages/amazon-cognito-identity-js/webpack.config.js:1]: Nit: Is this required?

File: packages/amazon-cognito-identity-js/webpack.config.js [hunk 5]
```diff
@@ -28,27 +27,28 @@ var config = {
...
-		}),
-		new CompressionPlugin({
-			include: /\.min\.js$/,
-		}),
+		new webpack.BannerPlugin({ banner, raw: true })
 	],
+	optimization: {
+		minimizer: [
+			new TerserPlugin({
+				sourceMap: true,
```
Comment (cshfang) [inline packages/amazon-cognito-identity-js/webpack.config.js:35]: Looking at documentation, it seems like this property is supposed to be under `terserOptions` - but it seems to be working here...?

**Review by cshfang — COMMENTED** [review]
Just had some questions about the config. Otherwise looks 🔥

File: packages/amazon-cognito-identity-js/webpack.config.js (hunk 4 above)
Comment (jimblanc) [reply packages/amazon-cognito-identity-js/webpack.config.js:1]: Not technically required but it makes it a little easier to determine which version of the plugin corresponds to the version of Webpack used. I can remove if this is just adding more confusion 😓

File: packages/amazon-cognito-identity-js/webpack.config.js (hunk 5 above)
Comment (jimblanc) [reply packages/amazon-cognito-identity-js/webpack.config.js:35]: Are you looking at the docs for the correct version of the plugin? https://www.npmjs.com/package/terser-webpack-plugin/v/1.4.3

File: packages/amazon-cognito-identity-js/webpack.config.js [hunk 6]
```diff
@@ -28,27 +27,28 @@ var config = {
...
+				terserOptions: {
+					compress: true
```
[the remark this answers is not in the document]
Comment (jimblanc) [reply packages/amazon-cognito-identity-js/webpack.config.js:37]: I believe it defaults to `{}` which I think means enabled. Just added in this PR to make it explicit considering the removal of the `CompressionPlugin`.

File: packages/amazon-cognito-identity-js/webpack.config.js (hunk 4 above)
Comment (cshfang) [reply packages/amazon-cognito-identity-js/webpack.config.js:1]: It's not a big deal. Just seems like an easy maintenance miss

File: packages/amazon-cognito-identity-js/webpack.config.js (hunk 5 above)
Comment (cshfang) [reply packages/amazon-cognito-identity-js/webpack.config.js:35]: Ok figured it was a version thing. The documentation on the Webpack site is a little different

**Review by cshfang — APPROVED** [review]

**Pull request merged**